pyomo/core/base/expression.py

```diff
@@ -235,7 +193,17 @@ def __ipow__(self, other):
         return numeric_expr._pow_dispatcher[e.__class__, other.__class__](e, other)
 
 
-class _GeneralExpressionData(_GeneralExpressionDataImpl, ComponentData):
+class _ExpressionData(metaclass=RenamedClass):
+    __renamed__new_class__ = NamedExpressionData
+    __renamed__version__ = '6.7.2.dev0'
+
+
+class _GeneralExpressionDataImpl(metaclass=RenamedClass):
+    __renamed__new_class__ = NamedExpressionData
+    __renamed__version__ = '6.7.2.dev0'
+
+
+class ExpressionData(NamedExpressionData, ComponentData):
```

Comment:
I mildly prefer `ExpressionDataBase` just because we sometimes refer to big-E Expressions as named Expressions and so I found that confusing when it was a class name for the base class.

Reply:
I was in the middle of making the change from `NamedExpressionData` -> `ExpressionDataBase`, when I realized that one of the things we do in that class is to override:
```python
def is_named_expression_type(self):
    """A boolean indicating whether this in a named expression."""
    return True
```
Given that, and the fact that `ExpressionData` and `ObjectiveData` are both special types of "named expressions", I think I may go back to advocating that we leave things as-is.